From 5347573a6e3664be2666f2870d9ac5457f47ff20 Mon Sep 17 00:00:00 2001
From: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
Date: Fri, 20 Mar 2026 14:16:07 -0300
Subject: [PATCH] csi-powerscale: add proxy-authz-token template

This patch consists of adding proxy-authz-token in the helm chart
templates directory. It's needed because this helm chart needs a config
secret that should be created manually.

Signed-off-by: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
---
 charts/csi-isilon/templates/controller.yaml         |  4 ++--
 charts/csi-isilon/templates/node.yaml               |  4 ++--
 charts/csi-isilon/templates/proxy-authz-tokens.yaml | 10 ++++++++++
 3 files changed, 14 insertions(+), 4 deletions(-)
 create mode 100644 charts/csi-isilon/templates/proxy-authz-tokens.yaml

diff --git a/charts/csi-isilon/templates/controller.yaml b/charts/csi-isilon/templates/controller.yaml
index 0a16220..1f4137e 100644
--- a/charts/csi-isilon/templates/controller.yaml
+++ b/charts/csi-isilon/templates/controller.yaml
@@ -508,12 +508,12 @@ spec:
             - name: ACCESS_TOKEN
               valueFrom:
                 secretKeyRef:
-                  name: proxy-authz-tokens
+                  name: {{ .Release.Name }}-proxy-authz-tokens
                   key: access
             - name: REFRESH_TOKEN
               valueFrom:
                 secretKeyRef:
-                  name: proxy-authz-tokens
+                  name: {{ .Release.Name }}-proxy-authz-tokens
                   key: refresh
           volumeMounts:
             {{- if $authorizationConfigSecret }}
diff --git a/charts/csi-isilon/templates/node.yaml b/charts/csi-isilon/templates/node.yaml
index caca236..b56da5c 100644
--- a/charts/csi-isilon/templates/node.yaml
+++ b/charts/csi-isilon/templates/node.yaml
@@ -295,12 +295,12 @@ spec:
             - name: ACCESS_TOKEN
               valueFrom:
                 secretKeyRef:
-                  name: proxy-authz-tokens
+                  name: {{ .Release.Name }}-proxy-authz-tokens
                   key: access
             - name: REFRESH_TOKEN
               valueFrom:
                 secretKeyRef:
-                  name: proxy-authz-tokens
+                  name: {{ .Release.Name }}-proxy-authz-tokens
                   key: refresh
           volumeMounts:
             {{- if $authorizationConfigSecret }}
diff --git a/charts/csi-isilon/templates/proxy-authz-tokens.yaml b/charts/csi-isilon/templates/proxy-authz-tokens.yaml
new file mode 100644
index 0000000..38f35cf
--- /dev/null
+++ b/charts/csi-isilon/templates/proxy-authz-tokens.yaml
@@ -0,0 +1,10 @@
+{{- if or .Values.authorization.enabled .Values.csm.authorization }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Release.Name}}-proxy-authz-tokens
+type: Opaque
+data:
+  access: {{ .Values.proxyAuthzTokens.access |  default " " | quote}}
+  refresh: {{ .Values.proxyAuthzTokens.refresh | default " " | quote }}
+{{- end }}
-- 
2.34.1

